libvast_test/vast/test/test.hpp

```diff
@@ -143,6 +143,13 @@ struct less_equal_compare {
 
 namespace vast::test {
 
+template <class T>
+T unbox(std::optional<T> x) {
+  if (!x)
+    CAF_FAIL("x == none");
+  return std::move(*x);
+}
```

Comment:
Not that performance matters much in unit tests, but there should probably be three overloads for this: Two that take either a `std::optional<T>& x` and `std::optional<T>&& x` that move the result out, and one that takes a `const std::optional<T>& x` that does not.

Reply:
Why three as opposed to two? Moving out of a mutable reference sounds unusual?

Reply:
Just added the additional overloads 👍

Reply:
@lava's point is kinda fair, but it depends on the expected semantics of `unbox`—does it always move internally?

Reply:
In `CAF`, they just have a value-based overload:

```
template <class T>
T unbox(caf::optional<T> x) {
  if (!x)
    CAF_FAIL("x == none");
  return std::move(*x);
}
```

That's why I just provided this one function template originally. I don't have strong preferences on multiple overloads. I like the simplicity of just always moving the value and taking a copy.

Reply:
I'll leave this up to @lava to decide. I have no strong opinion on this.

Reply:
I don't care strongly, we're not going to re-use the boxed value in test code anyways. I just thought maybe I missed something that required a third overload.